pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -122,6 +122,7 @@ public static PinotQuery compileToPinotQuery(String sql)
 
     // Extract OPTION statements from sql as Calcite Parser doesn't parse it.
     List<String> options = extractOptionsFromSql(sql);
+    sql = removeCommentedOptionsFromSql(sql);
```

Comment:
Why do we need to remove the commented options again here?

I feel a simpler and more readable way to support this is to first remove the commented part from the sql, then extract the options

Reply:
It is not wise to remove the commented part from the sql because it may distort the query. Ex: 
`SELECT * FROM tableA where colA LIKE '%--%'` becomes `SELECT * FROM tableA where colA LIKE '%%'`

But this is the change I have made – The first thing that will happen is the removal of commented options. After this, query parsing will proceed like before.